File: Arrays & Hashing/1929. Concatenation of Array/Solution.py
from typing import List

class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        # N : len(nums)
        # Time Complexity: O(N)
        # Space Complexity: O(2N)
        n = len(nums)
        for i in range(n):
            nums.append(nums[i])
        return nums

File: Arrays & Hashing/1929. Concatenation of Array/Solution.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        // N : len(nums)
        // Time Complexity: O(N)
        // Space Complexity: O(2N)
        int n = nums.length;
        int[] res = new int[n * 2];
        for (int i = 0; i < (n * 2); i++) {
            res[i] = nums[i % n];
        }
        return res;
    }
}